gammapy/modeling/models/temporal.py

```diff
@@ -334,6 +334,74 @@ def integral(self, t_min, t_max, **kwargs):
         return integral / self.time_sum(t_min, t_max)
 
 
+class GeneralizedGaussianTemporalModel(TemporalModel):
+    r"""A generalized Gaussian temporal profile
+
+    ..math::
+            F(t) = exp( - (\frac{|t - t_{ref}|}{\sigma_rise}) ^ nu)   for  t < t_ref
+            
+            F(t) = exp( - (\frac{|t - t_{ref}|}{\sigma_decay}) ^ nu)   for  t > t_ref            
+
+    Parameters
+    ----------
+    t_ref: `~astropy.units.Quantity`
+        The time of the pulse's maximum intensity.
+    sigma_rise : `~astropy.units.Quantity`
+        Rise time constant.
+    sigma_decay : `~astropy.units.Quantity`
+        Decay time constant.
+    nu : `~astropy.units.Quantity`
+        Pulse sharpness -> lower values implies a more peaked pulse
+    
+    """
+
+    tag = ["GeneralizedGaussianTemporalModel", "gengauss"]
+
+    _t_ref_default = Time("2000-01-01")
+    t_ref = Parameter("t_ref", _t_ref_default.mjd, unit = "day", frozen=False)
+    sigma_rise = Parameter("sigma_rise", "1d", frozen=False)
+    sigma_decay = Parameter("sigma_decay", "1d", frozen=False)
+    nu = Parameter("nu", 2, unit = "", frozen=False)
```

Comment:
For consistency with the [spatial gen-gauss](https://docs.gammapy.org/dev/api/gammapy.modeling.models.GeneralizedGaussianSpatialModel.html) the shape parameter could be eta=1/nu